### DISCUSSION TITLE
Modal Footer component request

### DISCUSSION THREAD

**Discussion by FakieSui** [mantinedev/mantine#1141] [Feature requests]
First of all, thanks for your wonderful work @rtivital, Mantine is from far my favorite react lib!

I'm opening this discussion to request a ModalFooter that would be similar to Material's. Indeed, it can be really useful to keep a button accessible along a scrollable Modal

Let me know if it was planned or if the idea is interesting,
Antoine.

**Comment by rtivital** [comment]
Thanks for kind words! We will consider adding this feature.

**Reply by SubJunk** [reply]
I would also really like this feature. I'm developing a file/folder chooser component and it's quite hard to do without a way to have a choose/select button always on screen

**Reply by mohammad-zr** [reply]
Hi @rtivital
Is this feature available? I also need it but don't see in documentation

**Reply by rtivital** [reply]
No, this feature is not available

**Reply by Akronae** [reply]
Would be useful for me too, when using a form inside a modal.

**Comment by simonschwitzky** [comment]
Espeically with longer forms that require scrollable content, a fixed modal footer is super useful to always show clear submit and cancel buttons.

**Comment by Sajarin-M** [comment]
is it planned in v7 ?

**Reply by rtivital** [reply]
No, it is not planned to be added to the library

**Comment by BenSilver1337** [comment]
The implementation of a ModalFooter component similar to Material's can be a valuable addition to a UI library like Mantine. A footer that stays fixed or remains accessible while scrolling within a modal can enhance user experience, especially when dealing with longer modal content.

**Comment by btmnk** [comment]
This worked for us just fine as styles for a group with buttons:

```css
.footerContainer {
  position: sticky;
  bottom: 0;
  left: 0;
  right: 0;
  background: white;
  z-index: 1;
}
```

(I know it's an old discussion but it popped up first when googling for a modal footer in mantine and maybe this helps others)

**Comment by lorkel** [comment]
Also found this from a search for a sticky footer. Here's an implementation using the Styles API + Tailwind CSS that is working for us:

```
<Modal 
  opened={opened} 
  onClose={close} 
  title="Some Title"
  classNames={{
    root: 'h-full',
    content: 'flex flex-col',
    body: 'flex flex-col h-full overflow-hidden'
  }}
>
  <div className="flex-1 overflow-y-auto">
    {/* scrolling / long content here */}
  </div>
  <div>
    {/* the footer here */}
  </div>
</Modal>
```

**Comment by davidchalifoux** [comment]
+1 For wanting this to be available for both Modals and Drawers.

Would you accept a PR for this @rtivital ?

**Reply by rtivital** [reply]
No, currently, it is not planned for 7.x releases.

**Comment by tylim88** [comment]
my solution is 
```
<Modal
styles={{
   inner:{alignItems: 'flex-end' }
}}
>
</Modal>
```

**Comment by matejdostal** [comment]
Another option is to use Modal.Header component (as footer) with few styles redefined, like this:

`<Modal.Header style={{ bottom: 0, borderBottom: 0, borderTop: "1px solid var(--mantine-color-default-border)" }}>
    ...
<Modal.Header>`

**Reply by Akronae** [reply]
this is the best solution! thanks
```jsx
<Modal.Header pos={"sticky"} bottom={0}>
</Modal.Header>
```

**Reply by Crocsx** [reply]
Yeah doing same here

```
<Modal.Header
  component={'footer'}
  pos={'sticky'}
  bottom={0}
></Modal.Header>
```

I guess a PR to main repo would not take to long if the goal just to copy this Header. Maybe will take a look in the week if this solution is OK.

**Comment by zavbala** [comment]
Well I came here looking for a drawer solution, this is mine, it's working

```jsx

const DefaultProps = {
  offset: 16,
  size: 'lg',
  radius: 'md',
  padding: 'xl',
  position: 'right' as any,
};

const Flap = ({ footer, children, ...self }: Props) => {
  const isMobile = useMediaQuery('(max-width: 820px)');

  return (
    <Drawer.Root
      {...DefaultProps}
      styles={{ content: { overflow: 'hidden' } }}
      {...self}
    >
      <Drawer.Overlay />

      <Drawer.Content>
        <Drawer.Header h="5rem">
          <Drawer.Title>{self.title}</Drawer.Title>
          <Drawer.CloseButton />
        </Drawer.Header>

        <Drawer.Body
          pb={0}
          h="calc(100vh - (var(--drawer-offset) * 2) - 10rem)"
        >
          <ScrollArea h="100%">{children}</ScrollArea>
        </Drawer.Body>

        {footer && (
          <Drawer.Header h="5rem" component="footer" pos="sticky" bottom={0}>
            {footer}
          </Drawer.Header>
        )}
      </Drawer.Content>
    </Drawer.Root>
  );
};

```

The idea is to preserve the `footer` sticky as header does, and only scrolls into `body`no content itself.
